numba/tests/test_jit_module.py

```diff
@@ -0,0 +1,155 @@
+            lines = source_lines.format(jit_options=jit_options)
+            f.write(lines)
+        # Add test_module to sys.path so it can be imported
+        sys.path.insert(0, tempdir)
+        test_module = importlib.import_module(temp_module_name)
+        yield test_module
+        sys.modules.pop(temp_module_name, None)
+        sys.path.remove(tempdir)
+        shutil.rmtree(tempdir)
+
+    def test_jit_module(self):
+        with self.create_temp_jitted_module() as test_module:
+            self.assertTrue(isinstance(test_module.inc, dispatcher.Dispatcher))
+            self.assertTrue(isinstance(test_module.add, dispatcher.Dispatcher))
+            self.assertTrue(isinstance(test_module.inc_add, dispatcher.Dispatcher))
+            self.assertTrue(test_module.mean is np.mean)
+            self.assertTrue(inspect.isclass(test_module.Foo))
+
+            # Test output of jitted functions is as expected
+            x, y = 1.7, 2.3
+            self.assertEqual(test_module.inc(x), test_module.inc.py_func(x))
+            self.assertEqual(test_module.add(x, y), test_module.add.py_func(x, y))
+            self.assertEqual(test_module.inc_add(x), test_module.inc_add.py_func(x))
+
+    def test_jit_module_jit_options(self):
+        jit_options = {"nopython": True,
+                       "nogil": False,
+                       "error_model": "numpy",
+                       }
+        with self.create_temp_jitted_module(**jit_options) as test_module:
+            self.assertTrue(test_module.inc.targetoptions == jit_options)
+
+    def test_jit_module_jit_options_override(self):
+        source_lines = """
+from numba import jit, jit_module
+
+@jit(nogil=True, forceobj=False)
+def inc(x):
+    return x + 1
+
+def add(x, y):
+    return x + y
+
+jit_module({jit_options})
+"""
+        jit_options = {"nopython": True,
+                       "error_model": "numpy",
+                       }
+        with self.create_temp_jitted_module(source_lines=source_lines, **jit_options) as test_module:
+            self.assertTrue(test_module.add.targetoptions == jit_options)
+            # Test that manual jit-wrapping overrides jit_module options
+            self.assertTrue(test_module.inc.targetoptions == {'nogil': True, 'forceobj': False})
+
+    def test_jit_module_logging(self):
+        logger = logging.getLogger('numba.decorators')
+        logger.setLevel(logging.DEBUG)
+        jit_options = {"nopython": True,
+                       "error_model": "numpy",
+                       }
+        with self.capture:
+            with self.create_temp_jitted_module(**jit_options) as test_module:
+                logs = self.capture.getvalue()
+                expected = ["Auto decorating function",
+                            "from module {}".format(test_module.__name__),
+                            "with jit and options: {}".format(jit_options)]
+                assert all(i in logs for i in expected)
```

Comment:
Please use `self.assert*` functions to be consistent with the rest of the test suite (plain `unittest` style is in use). I think `self.assertTrue` would likely be appropriate here?

Reply:
great, thanks